iothub/device/src/Pipeline/MqttTransportHandler.cs

```diff
@@ -1181,120 +1176,100 @@ private void HandleTwinResponse(MqttApplicationMessageReceivedEventArgs received
             {
                 byte[] payloadBytes = receivedEventArgs.ApplicationMessage.Payload ?? Array.Empty<byte>();
 
-                if (_getTwinResponseCompletions.TryRemove(receivedRequestId, out TaskCompletionSource<GetTwinResponse> getTwinCompletion))
+                if (_pendingTwinOperations.TryRemove(receivedRequestId, out PendingMqttTwinOperation twinOperation))
```

Comment:
This "if" statement is the same as the "else if" statement on line 1252

Reply:
They should probably be consolidated in some way

Reply:
Something like

```csharp
                if (_pendingTwinOperations.TryRemove(receivedRequestId, out PendingMqttTwinOperation twinOperation))
{
    if (twinOperation.TwinPatchTask != null)
    {
        // handle twin patch deserialization/error handling    
    }
    else if (twinOperation.TwinResponseTask != null)
    {
        // handle get twin response deserialization/error handling    
    }
}
```

Reply:
Fixed